shared/crypto/output/index.tsx

```diff
@@ -42,44 +42,67 @@ export const SignedSender = (props: SignedSenderProps) => {
   const waiting = Container.useAnyWaiting(waitingKey)
   // Store
   const signed = Container.useSelector(state => state.crypto[operation].outputSigned)
-  const signedBy = Container.useSelector(state => state.crypto[operation].outputSender)
+  const signedByUsername = Container.useSelector(state => state.crypto[operation].outputSenderUsername)
+  const signedByFullname = Container.useSelector(state => state.crypto[operation].outputSenderFullname)
   const outputStatus = Container.useSelector(state => state.crypto[operation].outputStatus)
 
-  const canSelfSign = operation === Constants.Operations.Encrypt || operation === Constants.Operations.Sign
+  const isSelfSigned = operation === Constants.Operations.Encrypt || operation === Constants.Operations.Sign
+  const avatarSize = isSelfSigned ? 16 : 48
+  const usernameType = isSelfSigned ? 'BodySmallBold' : 'BodyBold'
 
   if (!outputStatus || (outputStatus && outputStatus === 'error')) {
     return null
   }
 
   return (
     <Kb.Box2 direction="horizontal" fullWidth={true} alignItems="center" style={styles.signedContainer}>
-      <Kb.Box2 direction="horizontal" gap="xtiny" alignItems="center" style={styles.signedSender}>
-        {signed && signedBy
-          ? [
-              <Kb.Avatar key="avatar" size={16} username={signedBy.stringValue()} />,
-              <Kb.Text key="signedBy" type="BodySmall">
-                Signed by {canSelfSign ? ' you, ' : ''}
+      {signed && signedByUsername ? (
```

Comment:
This jsx is hard to follow, I wonder if it could be factored into separate components some. No need to change now.

Reply:
Yeah you're right. I'll clean it up and then merge. thanks!